astropy/units/utils.py

```diff
@@ -51,7 +51,7 @@ def _get_first_sentence(s: str) -> str:
 
 
 def _iter_unit_summary(
-    namespace: dict[str, object],
+    namespace: Mapping[str, object],
```

Comment:
One big difference between using `Mapping` and `dict` in parameter annotations is that `Mapping` makes the promise that the argument is not mutated. Furthermore, [one of the consequences of `dict` being mutable is that it is invariant](https://peps.python.org/pep-0483/#covariance-and-contravariance), so if a parameter is annotated with `dict` then type checkers must reject passing in a `dict` with a narrower value type.